### PR TITLE
Task/module 8

### PR DESCRIPTION
**Pull request by AB619**
1. Vue Router added.
2. Routes added -> /movies , /movies/:id, /movies?search="abc"

### CODE REVIEW

File: src/components/Banner.vue [hunk 1]
```diff
@@ -25,7 +38,7 @@ const clickHandler = () => {
         <span>roulette</span>
       </div>
       <div v-if="movieStore.isMovieDetailOpen" class="header__right">
-        <img src="../assets/search.svg" @click="movieStore.toggleMovieDetailPanel(null)">
+        <img src="../assets/search.svg" @click="prevIconClickHandler">
```
Comment (hicugi) [inline src/components/Banner.vue]: Use `RouterLink` component to go back to main page

File: src/components/MovieList.vue [hunk 2]
```diff
@@ -21,14 +21,14 @@ const clickHandler = (id: number) => {
 <template>
   <div class="movies">
     <div v-if="movies.length > 0" class="movies__list">
-      <div v-for="(m) in movies" class="card" :key="m.id" @click="clickHandler(m.id)">
+      <RouterLink v-for="(m) in movies" class="card" :key="m.id" @click="clickHandler(m.id)" :to="`/movies/${m.id}`">
```
Comment (hicugi) [inline src/components/MovieList.vue]: 1. I don't think that you need the `click` event
2. Use [named routes](https://router.vuejs.org/guide/essentials/named-routes.html) for the link

File: src/views/NotFoundView.vue [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<script setup>
+import Footer from "../components/Footer.vue";
+</script>
+
+<template>
+    <div class="not__found">
+        <h3>404 Page Not Found</h3>
+    </div>
+    <Footer />
```
Comment (hicugi) [inline src/views/NotFoundView.vue:9]: You can create & use `Layout` component for the `header` & `footer`, and past content in it

**Review by hicugi — COMMENTED** [review]
There is a room for improvements

File: src/components/Banner.vue (hunk 1 above)
Comment (AB619) [reply src/components/Banner.vue]: fixed

File: src/components/MovieList.vue (hunk 2 above)
Comment (AB619) [reply src/components/MovieList.vue]: fixed

**Pull request merged**